map/extrapolation_benchmark/extrapolation_benchmark.cpp

```diff
@@ -0,0 +1,326 @@
+#include "map/extrapolation/extrapolator.hpp"
+
+#include "routing/base/followed_polyline.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+#include "geometry/latlon.hpp"
+#include "geometry/mercator.hpp"
+#include "geometry/polyline2d.hpp"
+
+#include "platform/location.hpp"
+
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include <cmath>
+#include <cstdint>
+#include <fstream>
+#include <limits>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "3party/gflags/src/gflags/gflags_declare.h"
+#include "3party/gflags/src/gflags/gflags.h"
+
+// This tool is written to estimate quality of location extrapolation. To launch the benchmark
+// you need tracks in csv file with the format described below. To generate the csv file
+// you need to go through following steps:
+// * take logs from "trafin" project production in gz files
+// * extract them (gzip -d)
+// * run track_analyzer tool with unmatched_tracks command to generate csv
+//   (track_analyzer -cmd unmatched_tracks -in ./trafin_log.20180517-0000)
+// * run this tool with csv_path equal to path to the csv
+//   (extrapolation_benchmark -csv_path=trafin_log.20180517-0000.track.csv)
+
+DEFINE_string(csv_path, "",
+              "Path to csv file with user in following format: mwm id (string), aloha id (string), "
+              "latitude of the first coord (double), longitude of the first coord (double), "
+              "timestamp in seconds (int), latitude of the second coord (double) and so on.");
+
+using namespace extrapolation;
+using namespace location;
+using namespace routing;
+using namespace std;
+
+namespace
+{
+struct GpsPoint
+{
+  GpsPoint(double timestampS, double lat, double lon)
+      : m_timestampS(timestampS)
+      , m_lat(lat)
+      , m_lon(lon)
+  {
+  }
+
+  double m_timestampS;
+  // @TODO(bykoianko) Using LatLog type instead of two double should be considered.
+  double m_lat;
+  double m_lon;
+};
+
+class Expectation
+{
+public:
+  void Add(double value)
+  {
+    m_averageValue = (m_averageValue * m_counter + value) / (m_counter + 1);
```

Comment:
```c++
++m_counter;
m_mean += (value - m_mean) / m_counter;
```

Reply:
Thanks. It's shorter then written now:
```
  m_mean = m_mean + (value - m_mean) / static_cast<double>(m_counter + 1);
  ++m_counter;
```
And no big value after m_averageValue * m_counter. Let's put as you proposed.